lib/oli/authoring/editing/page_editor.ex

```diff
@@ -240,38 +240,46 @@ defmodule Oli.Authoring.Editing.PageEditor do
   # deleted state of the activity revision correct.
   defp resurrect_or_delete_activity_references(revision, change, project_slug) do
 
-    # Handle the case where this change does not include content
-    case Map.get(change, "content") do
-
-      nil -> {revision, []}
+    if Map.get(change, :deleted) do
```

Comment:
`If Map.get(change, :deleted, false) do`  will make this more robust.  This guarantees that the deletion code that follows only runs when the page is being edited to set its "deleted" field to "true".